fastapi_auth_jwt_demo/tests/test_fastapi_auth_jwt_demo.py

```diff
@@ -0,0 +1,146 @@
+            "fastapi_auth_jwt_demo.fastapi_endpoint_auth_jwt_demo"
+        )._handle_registry_sync()
+
+    def _get_token(self, aud=None, email=None):
+        validator = self.env["auth.jwt.validator"].search([("name", "=", "demo")])
+        payload = {
+            "aud": aud or validator.audience,
+            "iss": validator.issuer,
+            "exp": time.time() + 60,
+        }
+        if email:
+            payload["email"] = email
+        access_token = jwt.encode(
+            payload, key=validator.secret_key, algorithm=validator.secret_algorithm
+        )
+        return "Bearer " + access_token
+
+    def test_whoami(self):
+        """A end-to-end test with positive authentication and partner retrieval."""
+        partner = self.env["res.users"].search([("email", "!=", False)])[0]
+        token = self._get_token(email=partner.email)
+        resp = self.url_open(
+            "/fastapi_auth_jwt_demo/whoami", headers={"Authorization": token}
+        )
+        resp.raise_for_status()
+        whoami = resp.json()
+        self.assertEqual(whoami.get("name"), partner.name)
+        self.assertEqual(whoami.get("email"), partner.email)
+        self.assertEqual(whoami.get("uid"), self.env.ref("base.user_demo").id)
+        # Try again in a user session, it works because fastapi ignores the Odoo session.
+        self.authenticate("demo", "demo")
+        resp = self.url_open(
+            "/fastapi_auth_jwt_demo/whoami", headers={"Authorization": token}
+        )
+        self.assertEqual(whoami.get("name"), partner.name)
+        self.assertEqual(whoami.get("email"), partner.email)
+        self.assertEqual(whoami.get("uid"), self.env.ref("base.user_demo").id)
+
+    def test_whoami_cookie(self):
+        """A end-to-end test with positive authentication and cookie."""
+        partner = self.env["res.users"].search([("email", "!=", False)])[0]
+        token = self._get_token(email=partner.email)
+        resp = self.url_open(
+            "/fastapi_auth_jwt_demo/cookie/whoami", headers={"Authorization": token}
+        )
+        resp.raise_for_status()
+        whoami = resp.json()
+        self.assertEqual(whoami.get("name"), partner.name)
+        self.assertEqual(whoami.get("email"), partner.email)
+        self.assertEqual(whoami.get("uid"), self.env.ref("base.user_demo").id)
+        cookie = resp.cookies.get("demo_auth")
+        self.assertTrue(cookie)
+        # Try again with the cookie.
+        resp = self.url_open(
+            "/fastapi_auth_jwt_demo/cookie/whoami",
+            headers={"Cookie": f"demo_auth={cookie}"},
+        )
+        resp.raise_for_status()
+        whoami = resp.json()
+        self.assertEqual(whoami.get("name"), partner.name)
+        self.assertEqual(whoami.get("email"), partner.email)
+        self.assertEqual(whoami.get("uid"), self.env.ref("base.user_demo").id)
+        cookie = resp.cookies.get("demo_auth")
+        self.assertTrue(cookie)
+
+    def test_forbidden(self):
```

Comment:
@kobros-tech To solve the issue with the log you should decorate the method with a call to 'mute_logger' (`from odoo.tools import mute_logger`) to avoid the log of the stacktrace....
```suggestion
    @mute_logger('odoo.http')
    def test_forbidden(self):
```